torch/fx/experimental/symbolic_shapes.py

```diff
@@ -756,6 +771,27 @@ def __post_init__(self):
         if self.constraint_sizes is None:
             object.__setattr__(self, 'constraint_sizes', [None] * len(self.dynamic_sizes))
 
+
+@dataclass(frozen=True)
+class SubclassCreateSymbolicPolicy(CreateSymbolicPolicy):
+    """
+    The correct policy for a given inner tensor of a traceable tensor subclass
+    may differ from that of the outer policy. This structure allows for this
+    flexibility, with inner policies mapped via attr -> policy.
+    """
+    outer_policy: CreateSymbolicPolicy
+    inner_policies: Dict[str, CreateSymbolicPolicy]
+
+    # Convenience properties for call-sites that expect these.
+    @property
+    def dynamic_sizes(self):
+        return self.outer_policy.dynamic_sizes
+
+    @property
+    def constraint_sizes(self):
+        return self.outer_policy.constraint_sizes
```

Comment:
@voznesenskym please review

Reply:
Spoiler: this was wrong and hiding problems related to constraints being handled according to the outer tensor only. To get access to inner constraints, I addressed:
https://github.com/pytorch/pytorch/blob/6a86cf00adb071fe7200229cc1996dfefbb79289/torch/_dynamo/variables/builder.py#L1750

I could still use a review ensuring I did so properly